src/common/log.ts

```diff
@@ -49,6 +45,14 @@ export class Log {
     }
 
     /**
+     * Logs a warning message to the console.
+     */
+    public static logWarning(message: string, error?: any, outputChannel?: OutputChannel, logStack = true) {
+        // TODO: Refactor this code and create a better implementation
```

Comment:
Sounds good, but we should open a tracking issue and tag all TODOs with the issue number.